src/util/dbid.cpp

```diff
@@ -0,0 +1,16 @@
+#include "util/dbid.h"
+
+
+const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
+
+//static
+DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
+    if (variant.canConvert<value_type>() &&
+            variant.convert(kVariantType)) {
+        value_type value = variant.value<value_type>();
```

Comment:
Ah, right it is implicit shared and convert() only copies itself if the type differs. 
So this should work: 
```
if (variant.convert(kVariantType) ) {
    value_type value = variant.value<value_type>();
}
```
I think we need a comment to fix our findings for the future. Let me try: 

// "variant" is passed by value since we use the in place convert().
// Due to Qt's implicit sharing the value is only copied if the type id different. 
// The redundant conversion inside  value<value_type>() is bypassed since the type 
// already matches. We cannot use value<value_type>(), only because it returns the 
// valid id 0 in case of conversion errors.